src/geohash_helper.c

```diff
@@ -164,14 +164,15 @@ GeoHashRadius geohashCalculateAreasByShapeWGS84(GeoShape *shape) {
         geohashDecode(long_range, lat_range, neighbors.east, &east);
         geohashDecode(long_range, lat_range, neighbors.west, &west);
 
-        if (geohashGetDistance(longitude,latitude,longitude,north.latitude.max)
-            < radius_meters) decrease_step = 1;
-        if (geohashGetDistance(longitude,latitude,longitude,south.latitude.min)
-            < radius_meters) decrease_step = 1;
-        if (geohashGetDistance(longitude,latitude,east.longitude.max,latitude)
-            < radius_meters) decrease_step = 1;
-        if (geohashGetDistance(longitude,latitude,west.longitude.min,latitude)
-            < radius_meters) decrease_step = 1;
+        if (north.latitude.max < max_lat) 
+            decrease_step = 1;
+        if (south.latitude.min > min_lat) 
+            decrease_step = 1;
+        if (east.longitude.max < max_lon) 
+            decrease_step = 1;
+        if (west.longitude.min > min_lon)  
+            decrease_step = 1;
+
```

Comment:
```suggestion
        if (north.latitude.max < max_lat ||
            south.latitude.min > min_lat ||
            east.longitude.max < max_lon ||
            west.longitude.min > min_lon) {
            decrease_step = 1;
        }
```

Reply:
I've kept the style of the original code because it's  more easier to debug.   If we want to know which direction has problem, we can add debug statements like this:

  `if (north.latitude.max < max_lat) 
            decrease_step = 1,debug statement like print  "north has problem";
        if (south.latitude.min > min_lat) 
            decrease_step = 1,debug statement like print "south has problem";
        if (east.longitude.max < max_lon) 
            decrease_step = 1,debug statement like print "east has problem";
        if (west.longitude.min > min_lon)  
            decrease_step = 1,debug statement like print "west has problem";`

Your code is more efficient ( to judge  only once in the  best case), but it's not conducive to debugging.

In any case, the difference is not big. And I respect your choice.

I'll deal with the geo.ctl  as soon as possible.

Reply:
> Your code is more efficient ( to judge only once in the best case), but it's not conducive to debugging.

Agree with you, let us keep the old style.